### PR TITLE
Accept --license flag in `mc license register`

### PR DESCRIPTION
**Pull request by anjalshireesh**
## Community Contribution License

All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

If passed,

- validate and save it immediately in subnet config
- extract api key from it, and use for online registration with SUBNET (if `--airgap` is not passed)
- in case online registration with SUBNET fails for any reason, fallback to airgap mode (show URL to be visited on SUBNET)

## Motivation and Context

As SUBNET has moved to account level licenses, it is possible to download license even before the cluster is registered.
We'll now allow this license file to be used for registering clusters. This will simplify airgap mode registration as user will not
have to come back and update the license on mc/minio side.

## How to test this PR?

- Download license file from SUBNET
- Run `mc license register <alias> --license </path/to/minio.license>`
- Verify that registration is succesful

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/license-register.go [hunk 1]
```diff
@@ -41,6 +42,10 @@ var licenseRegisterFlags = append([]cli.Flag{
 		Name:  "name",
 		Usage: "Specify the name to associate to this MinIO cluster in SUBNET",
 	},
+	cli.StringFlag{
+		Name:  "license",
+		Usage: "License of the account on SUBNET",
```
Comment (harshavardhana) [inline cmd/license-register.go:47]:
```suggestion
		Usage: "license of the account on SUBNET",
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/license-register.go (hunk 1 above)
Comment (anjalshireesh) [reply cmd/license-register.go]: done

**Review by shtripat — APPROVED** [review]
LGTM. Verified.

**Review by klauspost — APPROVED** [review]
LGTM

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**